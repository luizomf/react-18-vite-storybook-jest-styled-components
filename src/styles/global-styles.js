import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle``;
