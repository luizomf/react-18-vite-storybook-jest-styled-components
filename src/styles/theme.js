export const theme = {
  colors: {
    primary: 'red',
    secondary: 'blue',
  },
};
