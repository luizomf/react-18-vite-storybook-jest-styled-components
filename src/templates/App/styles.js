import styled from 'styled-components';

export const Container = styled.h1`
  background: ${({ theme }) => theme.colors.primary};
`;
