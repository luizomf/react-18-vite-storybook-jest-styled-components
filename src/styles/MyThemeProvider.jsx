import { ThemeProvider } from 'styled-components';
import { theme } from './theme';
import P from 'prop-types';
import { GlobalStyles } from './global-styles';

export const MyThemeProvider = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
};

MyThemeProvider.propTypes = {
  children: P.node.isRequired,
};
