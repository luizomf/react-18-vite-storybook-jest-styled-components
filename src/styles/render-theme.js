import { render } from '@testing-library/react';
import { MyThemeProvider } from './MyThemeProvider';

export const renderTheme = (children) => {
  return render(<MyThemeProvider>{children}</MyThemeProvider>);
};
