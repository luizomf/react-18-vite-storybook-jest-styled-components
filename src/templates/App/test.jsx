import { screen } from '@testing-library/react';
import { App } from '.';
import { renderTheme } from '../../styles/render-theme';

it('should test', () => {
  renderTheme(<App>Test</App>);
  expect(screen.getByRole('heading', { name: 'Test' })).toMatchSnapshot();
});
