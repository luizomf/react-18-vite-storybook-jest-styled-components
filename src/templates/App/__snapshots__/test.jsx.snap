// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test 1`] = `
.c0 {
  background: red;
}

<h1
  class="c0"
>
  Test
</h1>
`;
